scripts/common_cmdline.sh

```diff
@@ -0,0 +1,86 @@
+# ------------------------------------------------------------------------------
+# vim:ts=4:et
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2016-2019 solidity contributors.
+# ------------------------------------------------------------------------------
+
+FULLARGS="--optimize --ignore-missing --combined-json abi,asm,ast,bin,bin-runtime,compact-format,devdoc,hashes,interface,metadata,opcodes,srcmap,srcmap-runtime,userdoc"
+OLDARGS="--optimize --combined-json abi,asm,ast,bin,bin-runtime,devdoc,interface,metadata,opcodes,srcmap,srcmap-runtime,userdoc"
+function compileFull()
+{
+    local expected_exit_code=0
+    local expect_output=0
+    if [[ $1 = '-e' ]]
+    then
+        expected_exit_code=1
+        expect_output=1
+        shift;
+    fi
+    if [[ $1 = '-w' ]]
+    then
+        expect_output=1
+        shift;
+    fi
+    if [[ $1 = '-i' ]]
+    then
+        expect_output=2
+        shift;
+    fi
+    local minor=22
+    if [[ $1 = '-v' ]]
+    then
+        local version=$2
+        if (echo $version | grep -Po '(?<=0.4.)\d+' >/dev/null); then
+            minor=$(echo $version | grep -Po '(?<=0.4.)\d+')
+        fi
+        shift 2
+    fi
+
+    local args=$FULLARGS
+    if (( minor < 22 )); then
```

Comment:
Without checking the major?

Actually it's major - minor - patch, so this should be called "patch"

Reply:
`$(echo $version | grep -Po '(?<=0.4.)\d+')`
It is checked if it is actually 0.4.

Can do full checks. In docs there is no lower than 0.4.x

Reply:
Why not do this inside the nested if then?

Reply:
Yes, the lowest minor version is 4, but we also have 5 and 6 as minor with patch less than 22

Reply:
I moved it to nested if